src/core/simulator/physics_engine/physics_engine.h

```diff
@@ -45,7 +45,11 @@ namespace argos {
       SEmbodiedEntityIntersectionItem(const SEmbodiedEntityIntersectionItem& s_item) :
          IntersectedEntity(s_item.IntersectedEntity),
          TOnRay(s_item.TOnRay) {}
-
+      SEmbodiedEntityIntersectionItem& operator=(const SEmbodiedEntityIntersectionItem& s_item) {
```

Comment:
Isn't this the same as the default copy constructor? Why is it needed?

Reply:
It is needed because GCC issues the following warning without it:

`/opt/data/git/argos3/src/core/simulator/physics_engine/physics_engine.cpp:52:29: warning: implicitly-declared ‘constexpr argos::SEmbodiedEntityIntersectionItem& argos::SEmbodiedEntityInterse
ctionItem::operator=(const argos::SEmbodiedEntityIntersectionItem&)’ is deprecated [-Wdeprecated-copy]
   52 |             s_item = tData[i];   
      |                             ^                                               
In file included from /opt/data/git/argos3/src/core/simulator/physics_engine/physics_engine.cpp:8:
/opt/data/git/argos3/src/core/simulator/physics_engine/physics_engine.h:45:7: note: because ‘argos::SEmbodiedEntityIntersectionItem’ has user-provided ‘argos::SEmbodiedEntityIntersectionItem
::SEmbodiedEntityIntersectionItem(const argos::SEmbodiedEntityIntersectionItem&)’
   45 |       SEmbodiedEntityIntersectionItem(const SEmbodiedEntityIntersectionItem& s_item) :
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                           
/opt/data/git/argos3/src/core/simulator/physics_engine/physics_engine.cpp: In function ‘bool argos::GetClosestEmbodiedEntityIntersectedByRay(argos::SEmbodiedEntityIntersectionItem&, const ar
gos::CRay3&, argos::CEmbodiedEntity&)’:      
/opt/data/git/argos3/src/core/simulator/physics_engine/physics_engine.cpp:74:29: warning: implicitly-declared ‘constexpr argos::SEmbodiedEntityIntersectionItem& argos::SEmbodiedEntityInterse
ctionItem::operator=(const argos::SEmbodiedEntityIntersectionItem&)’ is deprecated [-Wdeprecated-copy]
   74 |             s_item = tData[i];                                          
`
If you declare either the copy constructor or operator=(), GCC infers that you want some kind of non-default behavior for both, and issues a warning if it ends up creating a default version of the one you don't declare. 

From the C++11 standard:

§ D.2 Implicit declaration of copy functions [depr.impldec]

The implicit definition of a copy constructor as defaulted is deprecated if the class has a user-declared copy assignment operator or a user-declared destructor. The implicit definition of a copy assignment operator as defaulted is deprecated if the class has a user-declared copy constructor or a user-declared destructor (15.4, 15.8). In a future revision of this International Standard, these implicit definitions could become deleted (11.4).

And also: https://en.cppreference.com/w/cpp/language/rule_of_three